metashare/api/views.py

```diff
@@ -264,7 +264,15 @@ def commit(self, request, pk=None):
             )
         commit_message = serializer.validated_data["commit_message"]
         desired_changes = serializer.validated_data["changes"]
-        scratch_org.queue_commit_changes(request.user, desired_changes, commit_message)
+        target_directory = serializer.validated_data["target_directory"]
+        if target_directory not in scratch_org.valid_target_directories:
```

Comment:
I think `valid_target_directories` is now a `dict`, so doesn't this need to check the nested lists under `valid_target_directories["source"]`, `valid_target_directories["pre"]`, `valid_target_directories["post"]`, and `valid_target_directories["config"]`?